wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessagePresenter.swift

```diff
@@ -204,27 +207,44 @@ final class MessagePresenter: NSObject {
 
     func viewController(forImageMessagePreview message: ZMConversationMessage, actionResponder delegate: MessageActionResponder) -> UIViewController? {
         guard Message.isImage(message),
-            message.imageMessageData != nil else {
-                return nil
+              message.imageMessageData != nil else {
+            return nil
         }
 
         return imagesViewController(for: message, actionResponder: delegate, isPreviewing: true)
     }
 
     // MARK: - Pass
 
-    func createAddPassesViewController(fileMessageData: ZMFileMessageData) -> PKAddPassesViewController? {
-        guard let fileURL = fileMessageData.fileURL,
-            let passData = try? Data.init(contentsOf: fileURL) else {
-                return nil
+    @MainActor
+    func openPassesViewController(fileMessageData: ZMFileMessageData) async {
+        guard PKAddPassesViewController.canAddPasses() else { return }
+
+        guard let viewController = try? await makePassesViewController(fileMessageData: fileMessageData) else {
+            return // ignore error
         }
 
-        guard let pass = try? PKPass.init(data: passData) else { return nil }
+        targetViewController?.present(viewController, animated: true)
+    }
 
-        if PKAddPassesViewController.canAddPasses() {
-            return PKAddPassesViewController(pass: pass)
-        } else {
-            return nil
+    @MainActor
+    func makePassesViewController(fileMessageData: ZMFileMessageData) async throws -> PKAddPassesViewController? {
+        let pass = try await loadPass(fileMessageData: fileMessageData)
+        return PKAddPassesViewController(pass: pass)
+    }
+
+    private func loadPass(fileMessageData: ZMFileMessageData) async throws -> PKPass {
+        try await withCheckedThrowingContinuation { continuation in
+            Task.detached(priority: .userInitiated) {
```

Comment:
```suggestion
            Task {
```

I'm not fully ware of the consequences but I have read from [documentation](https://developer.apple.com/documentation/swift/task/detached(priority:operation:)-3lvix) advices like "Don’t use a detached task if it’s possible".